roles/switch_config/tasks/ios_config.yml

```diff
@@ -14,11 +14,15 @@
   when: "'dhcp_relay' in switch_vars"
 
 - name: Set VLANs
-  cisco.ios.ios_vlan:
-    name: "vlan{{ item }}"
-    vlan_id: "{{ item }}"
-    state: present
-  loop: "{{ vlan_list }}"
+  vars:
+    vlan_config: |
```

Comment:
You can avoid the yaml conversion by simple tweaking the jinja part:

```yaml
  vars:
    vlan_config: |-
      [   
      {% for vlan in vlan_list %}
      - name: "vlan{{ vlan }}"
        vlan_id: "{{ vlan }}"
      {% endfor %}
      ]
```
With that in place you can directly pass the var like this:
```yaml
  cisco.ios.ios_vlans:
    config: "{{ vlan_config }}"
```

Or even simpler:
```yaml
  cisco.ios.ios_vlans:
    config: |-
      [   
      {% for vlan in vlan_list %}
      - name: "vlan{{ vlan }}"
        vlan_id: "{{ vlan }}"
      {% endfor %}
      ]
    state: merged

```

Reply:
I tried doing semilar things but for some reason the module failed to accept them, not sure if this is a specific issue to the modules but unfortunately I was unable to find a better way to implement it with the time I had to work on it
```
fatal: [nfv_private_sw02]: FAILED! => changed=false 
  invocation:
    module_args:
      config: []
      running_config: null
      state: merged
  msg: dictionary requested, could not parse JSON or key=value
```